Physical_Engine/Query/Geometry3D.cs

```diff
@@ -0,0 +1,166 @@
+/*
+ * This file is part of the Buildings and Habitats object Model (BHoM)
+ * Copyright (c) 2015 - 2021, the respective contributors. All rights reserved.
+ *
+ * Each contributor holds copyright over their respective contributions.
+ * The project versioning (Git) records all such contribution source information.
+ *                                           
+ *                                                                              
+ * The BHoM is free software: you can redistribute it and/or modify         
+ * it under the terms of the GNU Lesser General Public License as published by  
+ * the Free Software Foundation, either version 3.0 of the License, or          
+ * (at your option) any later version.                                          
+ *                                                                              
+ * The BHoM is distributed in the hope that it will be useful,              
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of               
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                 
+ * GNU Lesser General Public License for more details.                          
+ *                                                                            
+ * You should have received a copy of the GNU Lesser General Public License     
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+using System;
+using System.Linq;
+using System.Collections.Generic;
+using System.ComponentModel;
+using BH.oM.Reflection.Attributes;
+using BH.oM.Physical.Elements;
+using BH.oM.Geometry;
+using BH.oM.Physical.FramingProperties;
+using BH.Engine.Geometry;
+
+namespace BH.Engine.Physical
+{
+    public static partial class Query
+    {
+        /***************************************************/
+        /**** Public Methods                            ****/
+        /***************************************************/
+
+        [Description("Gets the 3d geometry from the framing element")]
+        [Input("Beam", "The input beam to get the Geometry3D out of, i.e. its extrusion with its cross section along its centreline.")]
+        [Output("3d", "The composite geometry representing the framing element")]
+        public static IGeometry Geometry3D(this Beam beam)
```

Comment:
Would have this working on IFramingELement instead, to cover both beams, columns and piles etc

Reply:
That's right! Will do! Thanks

Reply:
Done in https://github.com/BHoM/BHoM_Engine/pull/2308/commits/3da43a3a15fd10cbea501c99fa5bb2c2e65ac05a